Chapter-4-applying-tactical-domain-driven-design/README.adoc

```diff
@@ -40,29 +77,238 @@ IMPORTANT: It makes no sense to consider the Domain Model in typical CRUD module
 
 We introduce elements like:
 
-- Aggregates
-- Entities
-- Value Objects
-- Domain Events
+- <<aggregate-root,Aggregates>>
+- <<entity,Entities>>
+- <<value-object,Value Objects>>
+- <<domain-events,Domain Events>>
+
+=== Contracts Bounded Context Tactical Design Analysis
+
+image::Assets/design-level-event-storming.png[design level event storming]
+
+Above image is the result of the Event Storming Design Level workshop.
+
+Let's focus on business rules that we identified during the workshop.
+
+1. **Annex Can Only Start During Binding Contract Period**:
+   - An annex can only be attached if it falls within the active period of the ___Binding Contract___. This ensures that all extensions and modifications are valid within the contract's timeframe.
+
+2. **Annex Can Only Be Attached To Active Contracts**:
+   - The ___Binding Contract___ must be active, meaning it hasn't expired or been terminated. This rule prevents any modifications to contracts that are no longer valid.
+
+3. **Previous Annex Must Be Signed**:
+   - Any new annex can only be added if the previous annex has been signed. This maintains a clear and enforceable order of amendments, ensuring that no annex is added without proper authorisation.
+
+Let's take a look closer to this rule. __Previous Annex Must Be Signed__. This rule is connected with the relationship between the new and previous annex. This is invariant.
+
+NOTE: Invariant is a rule or condition that must always be true for a system to be considered in a valid state. It ensures the integrity and consistency of the domain model
+
+To enforce these business rules and maintain consistency, we need a robust way to protect invariant, especially in a concurrent environment.
+
+This is where the concept of an <<aggregate-root,Aggregate Root>> comes into play.
+
+That’s why the __Binding Contract__ <<entity,entity>> has to be promoted to <<aggregate-root,Aggregate Root>> that will guard the annexes invariants.
+
+image::Assets/aggregate_root_canvas.png[]
+
+IMPORTANT: Aggregate Root is pattern that solve specific problem. It is not a silver bullet. "Aggregate Root Canvas" will help you make this decision consciously and consider all essential aspects like purpose, invariants, concurency, performance. We are encouraging you to use it in your design process.
+
+Let's proceed with and anlysis and take a look on __Annex__. Annexes are a part of the __Binding Contract__ <<aggregate,aggregate>>. They have to be uniquely identified and encapsulate business logic. We will model it as an <<entity, entity>>.
+
+__Binding Contract__ has signature property which has business logic and is no requirement to be uniquely identified. Signature can be compared by its properties. We want to use the ubiquitous language, so we have chosen <<value-object, value object>> as building block to model this concept.
```

Comment:
Drop "We want to use the ubiquitous language" as the connection between value object and ubiquitous language is not that direct - IMHO it can really mess up the understanding